integration-testing.md

```diff
@@ -0,0 +1,606 @@
+### Running a Single Test within a Test Class
+
+You can run only a single test within a test class by specifying the test class together with the `--filter` argument
+and the name to select the test that you are currently developing:
+
+```bash
+cd dev/tests/integration
+../../../vendor/bin/phpunit --filter 'testOnlyThisOneIsExecuted' ../../../app/code/Acme/Example/Test/Integration/ExampleTest.php
+```
+
+<a href="#common-mistakes"></a>
+## Common Mistakes
+
+<a href="#could-not-read-files-specified-as-arguments"></a>
+### Could Not Read Files Specified as Arguments
+
+This error occurs if the integration tests are executed from the wrong directory.
+
+<a href="#could-not-read-dev-tests-integration-phpunit-xml"></a>
+### Could Not Read “dev/tests/integration/phpunit.xml”
+
+This error occurs if the integration tests are executed from a different directory than `dev/tests/integration`. To fix
+the issue, change to the `dev/tests/integration` directory, adjust any relative paths accordingly, and run the tests
+again.
+
+<a href="#unable-to-connect-to-mysql"></a>
+### Unable to Connect to MySQL
+
+The PHP interpreter must be able to connect to the test database. By default, this means the tests have to run on the
+same host as the MySQL server. This problem most commonly occurs during development with Vagrant or Docker, where the
+Magento database is running on a virtual machine. If the tests then are executed using a PHP interpreter on the host
+system, the database might not be accessible.
+
+The error usually looks something like this:
+
+```bash
+phpunit
+
+Expected log
+
+exception 'PDOException' with message 'SQLSTATE[HY000] [2002] No such file or directory' in /var/www/magento2/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php:129
+```
+
+There are many ways this problem can be resolved, but the easiest is to run the tests in the virtual machine as well.
+
+<a href="#running-integration-tests-in-phpstorm"></a>
+## Running Integration Tests in PhpStorm
+
+When writing new integration tests or during debugging, it is convenient to execute tests from within the PhpStorm IDE.
+Ensure you have prepared the integration test environment before starting.
+
+<a href="#creating-an-integration-test-run-configuration"></a>
+### Creating an Integration Test Run Configuration
+
+Setting up a run configuration for integration tests is very similar to creating a run configuration for unit tests. See
+Running Unit Tests in PhpStorm for instructions on creating a basic run configuration. Then, configure the integration
+test to use the configuration file.
+
+<a href="#using-the-integration-test-configuration-file"></a>
+### Using the Integration Test Configuration File
+
+The only difference in the run configuration is that the integration test `phpunit.xml.dist` or `phpunit.xml`
+configuration file from the `dev/tests/integration` directory must be selected.
+
+<a href="#integration-tests-file-structure"></a>
```

Comment:
I think that this section needs to be either revisited or removed:
- The `Magento` folder does not exist.
- `bootstrap.php` does not exist either.
- It is missing the `_files` folder.

My vote would be to remove this.